File: app/src/main/java/com/huynd/skyobserver/services/PricesAPI.java
package com.huynd.skyobserver.services;

import com.huynd.skyobserver.models.PricePerDayBody;
import com.huynd.skyobserver.models.PricePerDayResponse;
import com.huynd.skyobserver.models.cheapestflight.CheapestPricePerMonthBody;
import com.huynd.skyobserver.models.cheapestflight.CheapestPricePerMonthResponse;

import java.util.List;
import java.util.Map;

import io.reactivex.Observable;
import retrofit2.http.Body;
import retrofit2.http.HeaderMap;
import retrofit2.http.POST;
import retrofit2.http.Path;

/**
 * Created by HuyND on 8/6/2017.
 */

public interface PricesAPI {
    @POST("{carrier}/{srcPort}/{dstPort}")
    Observable<List<PricePerDayResponse>> getPricePerDay(@HeaderMap Map<String, String> headers,
                                                         @Body PricePerDayBody body,
                                                         @Path("carrier") String carrier,
                                                         @Path("srcPort") String srcPort,
                                                         @Path("dstPort") String dstPort);

    @POST("addon/prodash/getlist")
    Observable<List<CheapestPricePerMonthResponse>> getCheapestPricePerMonth(
            @HeaderMap Map<String, String> headers,
            @Body CheapestPricePerMonthBody body);
}

File: app/src/main/java/com/huynd/skyobserver/services/PricesAPI.kt
package com.huynd.skyobserver.services

import com.huynd.skyobserver.entities.PricePerDayBody
import com.huynd.skyobserver.entities.PricePerDayResponse
import com.huynd.skyobserver.entities.cheapestflight.month.MonthCheapestBody
import com.huynd.skyobserver.entities.cheapestflight.month.CheapestPricePerMonthResponse

import io.reactivex.Observable
import retrofit2.http.Body
import retrofit2.http.HeaderMap
import retrofit2.http.POST
import retrofit2.http.Path

/**
 * Created by HuyND on 8/6/2017.
 */

interface PricesAPI {
    @POST("sapi/getprices/{carrier}/{srcPort}/{dstPort}")
    fun getPricePerDay(
            @HeaderMap headers:  Map<String, String>,
            @Body body: PricePerDayBody,
            @Path("carrier") carrier: String,
            @Path("srcPort") srcPort: String,
            @Path("dstPort") dstPort: String) : Observable<List<PricePerDayResponse>> 

    @POST("addon/prodash/getlist")
    fun getCheapestPricePerMonth(
            @HeaderMap headers: Map<String, String>,
            @Body body: MonthCheapestBody) : Observable<List<CheapestPricePerMonthResponse>> 
}
